packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -166,79 +240,77 @@ export const ItemDetailsModal: React.FC<ItemDetailsModalProps> = ({
-    );
-
-    setAllocated(Math.floor(allocatedUnits / packSize));
-  };
-
-  const allocateQuantities = () => {
+  const allocateQuantities = (
+    newValue: number,
+    issuePackSize: number | null
+  ) => {
+    setValue('quantity', String(newValue));
     // if invalid quantity entered, don't allocate
-    if (quantity < 1 || Number.isNaN(quantity) || !summaryItem) {
+    if (newValue < 1 || Number.isNaN(newValue)) {
       return;
     }
-    // if the selected item has no batch rows, allocate all to the placeholder
-    if (batchRows.length === 0) {
-      setValue('placeholder', quantity);
-      setAllocated(quantity);
-      return;
+    // If there is only one batch row, then it is the placeholder.
+    // Assign all of the new value and short circuit.
+    if (batchRows.length === 1) {
+      setBatchRows(
+        issueStock(batchRows, 'placeholder', newValue * (issuePackSize || 1))
+      );
     }
 
     // calculations are normalised to units
-    let toAllocate = quantity * packSize;
-    let batchAllocation = 0;
-
-    batchRows.forEach(batch => {
-      batchAllocation = 0;
-      // skip bigger pack sizes
-      const validBatch =
-        batch.packSize <= packSize &&
-        batch.availableNumberOfPacks > 0 &&
-        !batch.onHold;
-
-      if (validBatch) {
-        const allocatedUnits = Math.min(
-          toAllocate,
-          batch.availableNumberOfPacks * batch.packSize
-        );
-
-        batchAllocation = Math.floor(allocatedUnits / batch.packSize);
-        toAllocate -= batchAllocation * batch.packSize;
-      }
+    let toAllocate = newValue * (issuePackSize || 1);
+
+    const newBatchRows = [...batchRows];
+    const validBatches = newBatchRows.filter(
+      ({ packSize, onHold, availableNumberOfPacks }) =>
+        (issuePackSize ? packSize === issuePackSize : true) &&
+        availableNumberOfPacks > 0 &&
+        !onHold
+    );
 
-      setValue(batch.id, batchAllocation);
-      setValue(`${batch.id}_total`, batchAllocation * batch.packSize);
+    validBatches.forEach(batch => {
+      const batchRowIdx = newBatchRows.findIndex(({ id }) => batch.id === id);
+      const batchRow = newBatchRows[batchRowIdx];
+      if (!batchRow) return null;
+
+      const availableUnits = batch.availableNumberOfPacks * batch.packSize;
+      const allocatedUnits = Math.min(toAllocate, availableUnits);
+      const allocatedNumberOfPacks = Math.floor(
+        allocatedUnits / batch.packSize
+      );
+
+      toAllocate -= allocatedUnits;
+
+      newBatchRows[batchRowIdx] = {
+        ...batchRow,
+        quantity: allocatedNumberOfPacks,
+      };
     });
 
-    // allocate remainder to placeholder
-    setValue('placeholder', toAllocate);
-    setAllocated(quantity);
+    const placeholderIdx = newBatchRows.findIndex(
+      ({ id }) => id === 'placeholder'
+    );
+    const placeholder = newBatchRows[placeholderIdx];
+
+    if (!placeholder) throw new Error('No placeholder within item editing');
```

Comment:
I mean, not ideal, but shouldn't be happening. Need to get that nice logging in place!